includes/class-amp-theme-support.php

```diff
@@ -275,6 +279,44 @@ public static function register_content_embed_handlers() {
 		return $embed_handlers;
 	}
 
+	/**
+	 * Add the comments template placeholder marker
+	 *
+	 * @param array $args the args for the comments list..
+	 * @return array Args to return.
+	 */
+	public static function amp_set_comments_walker( $args ) {
+		$amp_walker     = new AMP_Comment_Walker();
+		$args['walker'] = $amp_walker;
+		// @todo Make threaded work by setting 'data-update-time' of the parent first level parent with the last timestamp.
+		$args['max_depth'] = 1;
+
+		return $args;
+	}
+
+	/**
+	 * Adds the form submit success and fail templates.
+	 *
+	 * @param string $post_id The post ID if action is 'comment_form'.
```

Comment:
Instead of overloading the `add_amp_comment_form_templates` method, let's have two separate methods for before and after.